src/algorithms/reco/FarForwardNeutralsReconstruction.h

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2025 Sebouh Paul
+
+#pragma once
+#include <DD4hep/Detector.h>
+#include <Evaluator/DD4hepUnits.h>
+#include <algorithms/algorithm.h>
+#include <algorithms/geo.h>
+#include <edm4eic/ClusterCollection.h>
+#include <edm4eic/ReconstructedParticleCollection.h>
+#include <spdlog/logger.h>
+#include <gsl/pointers>
+#include <memory>
+#include <string>                                 // for basic_string
+#include <string_view>                            // for string_view
+#include <vector>
+
+#include "algorithms/interfaces/WithPodConfig.h"
+#include "algorithms/reco/FarForwardNeutralsReconstructionConfig.h"
+
+namespace eicrecon {
+
+using FarForwardNeutralsReconstructionAlgorithm = algorithms::Algorithm<
+   algorithms::Input<
+       const edm4eic::ClusterCollection
+    >,
+    algorithms::Output<
+      edm4eic::ReconstructedParticleCollection,
+      edm4eic::ReconstructedParticleCollection
+    >
+    >;
+    class FarForwardNeutralsReconstruction :
+       public FarForwardNeutralsReconstructionAlgorithm,
+       public WithPodConfig<FarForwardNeutralsReconstructionConfig> {
+       public:
+         FarForwardNeutralsReconstruction(std::string_view name)
+                  : FarForwardNeutralsReconstructionAlgorithm{name,
+                                        {"inputClustersHcal"},
+                  {"outputNeutrons", "outputGammas"},
+                                        "Merges all HCAL clusters in a collection into a neutron candidate and photon candidates "} {}
+
+         void init() final;
+         void process(const Input&, const Output&) const final;
+         double calc_corr(double Etot, const std::vector<double>&) const;
+         bool isGamma(const edm4eic::Cluster& cluster) const;
+    private:
+        std::shared_ptr<spdlog::logger> m_log;
```

Comment:
These need to be private (they can't be part of the algorithm's API)
```suggestion
    private:
         double calc_corr(double Etot, const std::vector<double>&) const;
         bool isGamma(const edm4eic::Cluster& cluster) const;

        std::shared_ptr<spdlog::logger> m_log;
```

Reply:
Also, super minor point: I'd suggest being consistent with camel-vs.-snake case (e.g. `calc_corr` vs. `isGamma`). It's small, but helps with readability!

Reply:
m_log should not be defined in the algorithm. All the methods like `debug()` are already inherited.